editoast/src/views/timetable/stdcm_request_payload.rs

```diff
@@ -0,0 +1,285 @@
+use chrono::DateTime;
+use chrono::Duration;
+use chrono::Utc;
+use editoast_models::DbConnection;
+use editoast_schemas::rolling_stock::LoadingGaugeType;
+use editoast_schemas::train_schedule::Comfort;
+use editoast_schemas::train_schedule::MarginValue;
+use editoast_schemas::train_schedule::PathItem;
+use editoast_schemas::train_schedule::PathItemLocation;
+use itertools::Itertools;
+use serde::Deserialize;
+use serde::Serialize;
+use utoipa::ToSchema;
+
+use crate::core::pathfinding::PathfindingInputError;
+use crate::core::stdcm::STDCMPathItem;
+use crate::core::stdcm::STDCMStepTimingData;
+use crate::error::Result;
+use crate::models::temporary_speed_limits::TemporarySpeedLimit;
+use crate::models::towed_rolling_stock::TowedRollingStockModel;
+use crate::models::work_schedules::WorkSchedule;
+use crate::models::List;
+use crate::views::path::path_item_cache::PathItemCache;
+use crate::views::path::pathfinding::PathfindingFailure;
+use crate::views::path::pathfinding::PathfindingResult;
+use crate::Retrieve;
+use crate::SelectionSettings;
+
+use super::stdcm::STDCMError;
+
+editoast_common::schemas! {
+    STDCMRequestPayload,
+    PathfindingItem,
+    StepTimingData,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone, ToSchema)]
+pub struct PathfindingItem {
+    /// The stop duration in milliseconds, None if the train does not stop.
+    pub duration: Option<u64>,
+    /// The associated location
+    pub location: PathItemLocation,
+    /// Time at which the train should arrive at the location, if specified
+    pub timing_data: Option<StepTimingData>,
+}
+
+/// Convert the list of pathfinding items into a list of path item
+pub fn convert_steps(steps: &[PathfindingItem]) -> Vec<PathItem> {
+    steps
+        .iter()
+        .map(|step| PathItem {
+            id: Default::default(),
+            deleted: false,
+            location: step.location.clone(),
+        })
+        .collect()
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone, ToSchema)]
+pub struct StepTimingData {
+    /// Time at which the train should arrive at the location
+    pub arrival_time: DateTime<Utc>,
+    /// The train may arrive up to this duration before the expected arrival time
+    pub arrival_time_tolerance_before: u64,
+    /// The train may arrive up to this duration after the expected arrival time
+    pub arrival_time_tolerance_after: u64,
+}
+
+/// An STDCM request
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, ToSchema)]
+pub struct STDCMRequestPayload {
```

Comment:
A bit more bike shed on the naming: we're in `views`, I'd say we don't need the `Payload` suffix to understand this struct refers to the content of the http request of the STDCM endpoint. 

Additionally, I'd rather have this module to be included in the `stdcm` module rather than `timetable` directly (see previous comment). In this case I'd argue that `Request` is a good enough name: both `stdcm/request.rs` and `stdcm::Request` are explanatory enough.

Just like for the `core` definitions and the `Core` prefix, `views::timetable::stcdm` is already providing a lot of context.

Wdyt?

Reply:
Thanks for the great feedback! I agree with your points. I'll go ahead and remove the `Payload` suffix and move the module to `stdcm`. `Request` sounds perfect, and the context is clear.